audformat/core/utils.py

```diff
@@ -538,7 +544,17 @@ def intersect(
         return pd.Index([])
 
     if len(objs) == 1:
-        return objs[0]
+        if is_filewise_index(objs[0]):
+            return filewise_index()
+        elif is_segmented_index(objs[0]):
+            return segmented_index()
+        elif isinstance(objs[0], pd.MultiIndex):
+            return pd.MultiIndex.from_arrays(
+                [[]] * objs[0].nlevels,
+                names=objs[0].names,
+            )
```

Comment:
This is an interesting case. If you use `pd.testing.assert_index_equal()` it will not raise an error if two empty MultiIndex objects have different dtypes. As they can get here as we don't set the dtype.
A user is able to see this by looking at the `dtypes` attribute.

So the question is if we should ensure the same dtype?
You are also not doing this for the empty `pd.Index([])` case after `else`.

Reply:
Maybe we can make use of this helper function, which I also proposed in https://github.com/audeering/audformat/pull/241#discussion_r932147625:

```python
def create_alike_index(index: pd.Index, data: typing.Tuple) -> pd.Index:
    if isinstance(index, pd.MultiIndex):
        index = set_index_dtypes(
            pd.MultiIndex.from_tuples(data, names=index.names),
            dict(index.dtypes),
        )
    else:
        index = pd.Index(data, name=index.name, dtype=index.dtype)
    return index
```

Reply:
Then you could replace the whole `if`-statement by a single line:
```python
return alike_index(objs[0], ())
```

Reply:
> You are also not doing this for the empty pd.Index([]) case after else.

That was a mistake, will add it and also add that helper function you propose.

Reply:
done, I added `_alike_index()` but for the moment without a `data` argument.

Reply:
You are still not setting the dtype for MultiIndex, is this on purpose?

Reply:
Ups, no :) Should be fixed now.